src/index.js

```diff
@@ -21,30 +23,85 @@ import {
 
 export default class ImgixClient {
   constructor(opts = {}) {
-    this.settings = { ...DEFAULT_OPTIONS, ...opts };
+    this._settings = { ...DEFAULT_OPTIONS, ...opts };
     // a cache to store memoized srcset width-pairs
     this.targetWidthsCache = {};
-    if (typeof this.settings.domain != 'string') {
-      throw new Error('ImgixClient must be passed a valid string domain');
+
+    validateDomainString(this._settings.domain);
+
+    if (this._settings.includeLibraryParam) {
+      const defaultLibraryParam = 'js-' + ImgixClient.version();
+      this._settings.libraryParam =
+        this._settings.libraryParam || defaultLibraryParam;
     }
+  }
 
-    if (DOMAIN_REGEX.exec(this.settings.domain) == null) {
-      throw new Error(
-        'Domain must be passed in as fully-qualified ' +
-          'domain name and should not include a protocol or any path ' +
-          'element, i.e. "example.imgix.net".',
-      );
+  static version() {
+    return VERSION;
+  }
+
+  get settings() {
+    return this._settings;
+  }
+  get domain() {
+    return this.settings.domain;
+  }
+
+  set domain(value) {
+    validateDomainString(value);
+    this.settings.domain = value;
```

Comment:
💯 nice catch.

Initially, I had a settings setter/getter but reverted to only having a getter to avoid someone trying to change the entire object outside the constructor. This was left over from that.

I think this getter needs to be removed, and the `this._settings` in the constructor needs to be changed to `this.settings`.

Reply:
Discussed offline. We're keeping the underscore.